datafusion/physical-expr/src/window/lead_lag.rs

```diff
@@ -151,6 +149,64 @@ impl WindowShiftEvaluator {
     }
 }
 
+// implement ignore null for evaluate_all
+fn evaluate_all_with_ignore_null(
+    array: &ArrayRef,
+    offset: i64,
+    default_value: &ScalarValue,
+    is_lag: bool,
+) -> Result<ArrayRef, DataFusionError> {
+    let valid_indices: Vec<usize> = (0..array.len())
```

Comment:
there is already a method for that `array.nulls()`

Reply:
yes, but in this case I want to get the valid indices?

Reply:
`valid_indices` is everything in array which is not in `array.nulls()`

so `valid_indices.len() == array.len() - array.nulls().len()`

Reply:
Thanks